Soumyadeep/calculator_infix.kt

```diff
@@ -0,0 +1,32 @@
+import java.util.*
+
+fun main(){
+    val sc = Scanner(System.`in`)
+    print("Enter 1st number : ")
+    val num1 = sc.nextFloat()
+    print("Enter 2nd number : ")
+    val num2 = sc.nextFloat()
+    print("What do you want to perform? ( + , - , * , / ) : ")
+    val choice = sc.next()
+    val result = {x:Float,y:Float,ch:String -> when(ch){

```

Comment:
import java.util.*

fun main() {

    print("Enter a number")
    val no1 = Scanner(System.`in`).nextFloat()
    print("Enter a number")
    val no2 = Scanner(System.`in`).nextFloat()
    print(no1 calculate no2)
}

infix fun Float.calculate(other:Float): Any {
    print("enter choice: ")
    val choice = Scanner(System.`in`).nextInt()
    val result = {a:Float,b:Float->
        when (choice) {
            1 -> a + b
            2 -> a - b
            3 -> a * b
            else -> 0.0000000000000F
        }
    }
    return(result(this,other))
}

how does it look ?